test/IntegrationTests/test_getallchats.py

```diff
@@ -0,0 +1,317 @@
+
+        chat_response = requests.post(chat_url, json=chat_data)
+        logger.info(f"OUTPUT: Chat creation response status: {chat_response.status_code}")
+
+        self.assertEqual(chat_response.status_code, 200)
+        chat_data = chat_response.json()
+        self.valid_chat_id = chat_data["chat_id"]
+        logger.info(f"OUTPUT: Received valid chat_id: {self.valid_chat_id}")
+        logger.info("=== setUp completed successfully ===\n")
+
+    def test_wrong_customer_guid_and_wrong_chat_id(self):
+        """Test case 1: Invalid customer_guid and chat_id"""
+        logger.info("=== Starting Test Case 1: Wrong customer_guid and wrong chat_id ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": "invalid_customer_guid",
+            "chat_id": "invalid_chat_id"
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertEqual(response.status_code, 404)
+        self.assertIn("no chats found for this customer and chat id", response.text.lower())
+        logger.info("=== Test Case 1 Completed ===\n")
+
+    def test_correct_customer_guid_and_wrong_chat_id(self):
+        """Test case 2: Valid customer_guid but invalid chat_id"""
+        logger.info("=== Starting Test Case 2: Correct customer_guid and wrong chat_id ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": "invalid_chat_id"
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertEqual(response.status_code, 404)
+        self.assertIn("no chats found for this customer and chat id", response.text.lower())
+        logger.info("=== Test Case 2 Completed ===\n")
+
+    def test_wrong_customer_guid_and_correct_chat_id(self):
+        """Test case 3: Wrong customer_guid and correct chat_id"""
+        logger.info("=== Starting Test Case 3: Wrong customer_guid and correct chat_id ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": "invalid_customer_guid",
+            "chat_id": self.valid_chat_id
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertEqual(response.status_code, 404)
+        logger.info("=== Test Case 3 Completed ===\n")
```

Comment:
@santhiya-mj you should verify the response for the text as well
self.assertIn("no chats found for this customer and chat id", response.text.lower())

Reply:
Done sir, I've added the assertion to verify the response text.